launch/launch/actions/execute_process.py

```diff
@@ -205,50 +144,9 @@ def __init__(
             Defaults to 'False'.
         :param: respawn_delay a delay time to relaunch the died process if respawn is 'True'.
         """
-        super().__init__(**kwargs)
-        self.__cmd = [normalize_to_list_of_substitutions(x) for x in cmd]
-        self.__name = name if name is None else normalize_to_list_of_substitutions(name)
-        self.__cwd = cwd if cwd is None else normalize_to_list_of_substitutions(cwd)
-        self.__env = None  # type: Optional[List[Tuple[List[Substitution], List[Substitution]]]]
-        if env is not None:
-            self.__env = []
-            for key, value in env.items():
-                self.__env.append((
-                    normalize_to_list_of_substitutions(key),
-                    normalize_to_list_of_substitutions(value)))
-        self.__additional_env: Optional[List[Tuple[List[Substitution], List[Substitution]]]] = None
-        if additional_env is not None:
-            self.__additional_env = []
-            for key, value in additional_env.items():
-                self.__additional_env.append((
-                    normalize_to_list_of_substitutions(key),
-                    normalize_to_list_of_substitutions(value)))
-        self.__shell = shell
-        self.__sigterm_timeout = normalize_to_list_of_substitutions(sigterm_timeout)
-        self.__sigkill_timeout = normalize_to_list_of_substitutions(sigkill_timeout)
-        self.__emulate_tty = emulate_tty
-        self.__prefix = normalize_to_list_of_substitutions(
-            LaunchConfiguration('launch-prefix', default='') if prefix is None else prefix
-        )
-        self.__output = os.environ.get('OVERRIDE_LAUNCH_PROCESS_OUTPUT', output)
-        self.__output_format = output_format
-
-        self.__log_cmd = log_cmd
-        self.__on_exit = on_exit
-        self.__respawn = respawn
-        self.__respawn_delay = respawn_delay
-
-        self.__process_event_args = None  # type: Optional[Dict[Text, Any]]
-        self._subprocess_protocol = None  # type: Optional[Any]
-        self._subprocess_transport = None
-        self.__completed_future = None  # type: Optional[asyncio.Future]
-        self.__shutdown_future = None  # type: Optional[asyncio.Future]
-        self.__sigterm_timer = None  # type: Optional[TimerAction]
-        self.__sigkill_timer = None  # type: Optional[TimerAction]
-        self.__stdout_buffer = io.StringIO()
-        self.__stderr_buffer = io.StringIO()
-
-        self.__executed = False
+        self.__executable = Executable(cmd=cmd, prefix=prefix, name=name, cwd=cwd, env=env,
```

Comment:
@roger-strain nit: it'd seem to me that `self.__executable` is redundant with `self.process_description`